ddtrace/settings/http.py

```diff
@@ -44,30 +46,36 @@ def trace_headers(self, whitelist):
             normalized_header_name = normalize_header_name(whitelist_entry)
             if not normalized_header_name:
                 continue
-            self._whitelist_headers.add(normalized_header_name)
+            # Empty tag is replaced by the default tag for this header:
+            #  Host on the request defaults to http.request.headers.host
+            self.header_tags.setdefault(normalized_header_name, "")
 
         # Mypy can't catch cached method's invalidate()
-        self.header_is_traced.invalidate()  # type: ignore[attr-defined]
+        self.header_tag_name.invalidate()  # type: ignore[attr-defined]
 
         return self
 
     @cachedmethod()
-    def header_is_traced(self, header_name):
```

Comment:
are we able to get rid of this API?

technically it is public

Reply:
I added back the public method so we do not break the compatibility.